add android NDK build

MoroccanMalinois opened the pull request:
**By submitting this pull-request, I confirm the following:**

- I have read and understood the [Contributing Guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.

*Place an X inside the bracket (or click the box in preview) to confirm*
- [X] I confirm.

---

*Enter your pull-request summary here*

anonimal commented on Makefile:
```diff
@@ -120,6 +120,10 @@ static: static-dependencies
 
+release-static-android: static-dependencies
+	mkdir -p $(build)
+	cd $(build) && $(cmake) $(cmake-debug) $(cmake-static) -D ANDROID=1 -D KOVRI_DATA_PATH="/data/local/tmp/.kovri" ../ && $(MAKE)
```
Did you need to manually copy resources to this data path or did `make install-resources` somehow get sent there (is `~/.kovri` symlinked)?

MoroccanMalinois replied:
i needed to copy the files. but i did it manually since i got permission denied for some symlinks in certificates. i still need to look at it

MoroccanMalinois commented:
libjingle has moved to webrtc (and that's 215 MB :) ). 
I took the last version of the files from [here](https://chromium.googlesource.com/external/webrtc).
I changed a bit the cmake files and mtu.cc to keep the 2 new files strictly identical to the original repo(u can md5sum to check), rebased and squashed.

MoroccanMalinois commented:
Nothing specific to kovri yet, just beginning my kovri journey :) The build process can be a little difficult for someone w/o previous fight with NDK because of the shitty default gnustl. I use this [dockerfile](https://github.com/MoroccanMalinois/dockers/blob/master/android/kovri.dockerfile) if u want to have an idea. I had a permission denied with copying the resources bc of certificates symlinks, but didn't seem logical, so i still need to look at it.

anonimal commented:
>I use this dockerfile if u want to have an idea.

Interesting, looks like fun 😄 Does it build against [openssl-1.0.2k](https://www.openssl.org/source/openssl-1.0.2k.tar.gz)? (I wont PR an update if it's known to be problematic).

MoroccanMalinois commented:
I never tried but shouldn't be problematic.
